options/options.go

```diff
@@ -204,10 +204,14 @@ func (terragruntOptions TerragruntOptions) GetContext() (result collections.IDic
 		result.Set(key, value.Value)
 	}
 
-	terragruntOptions.Context["Source"] = terragruntOptions.Source
-	terragruntOptions.Context["TerragruntConfigPath"] = terragruntOptions.TerragruntConfigPath
-	terragruntOptions.Context["WorkingDir"] = terragruntOptions.WorkingDir
-	result.Set("TerragruntOptions", terragruntOptions.Context)
+	context := make(map[string]interface{}, len(terragruntOptions.Context))
```

Comment:
This code was causing crashes (very rarely) because multiple threads were modifying terragruntOptions.Context

Reply:
isn't there some deep copy function?

Reply:
I only want a shallow copy and this is the way to do it in Go. Not much syntactic sugar

Reply:
I only want a shallow copy and this is the way to do it in Go. Not much syntactic sugar